### PR TITLE
Chore - Type couchdb get

### PR DESCRIPTION
**Pull request by adrinr**
## Description
Typing the `db.get` response to respect the T type. Currently we return `T | any`, that is the same as `any`.
Typed the few breaking changes to `<any>` as there are not existing types for these use cases

Dependant: https://github.com/Budibase/budibase-pro/pull/205

### CODE REVIEW

File: packages/server/src/api/controllers/datasource.ts [hunk 1]
```diff
@@ -433,7 +433,7 @@ export async function destroy(ctx: UserCtx) {
 
 export async function find(ctx: UserCtx) {
   const db = context.getAppDB()
-  const datasource = await db.get(ctx.params.datasourceId)
+  const datasource = await sdk.datasources.get(ctx.params.datasourceId)
```
Comment (Rory-Powell) [inline packages/server/src/api/controllers/datasource.ts:436]: Can remove the `db` above now?

File: packages/server/src/api/controllers/datasource.ts [hunk 2]
```diff
@@ -449,7 +449,7 @@ export async function query(ctx: UserCtx) {
 
 export async function getExternalSchema(ctx: UserCtx) {
   const db = context.getAppDB()
-  const datasource = await db.get(ctx.params.datasourceId)
+  const datasource = await sdk.datasources.get(ctx.params.datasourceId)
```
Comment (Rory-Powell) [inline packages/server/src/api/controllers/datasource.ts:452]: Same as above

**Review by Rory-Powell — APPROVED** [review]
Nice work on this. Removing the fallback `any` is probably the right choice, it'll make us be more strict with properly typing db reads which I'm all for

**Comment by adrinr** [comment]
> Nice work on this. Removing the fallback `any` is probably the right choice, it'll make us be more strict with properly typing db reads which I'm all for

This will eventually "force" us to move the actual db.gets into the sdk or db functions, where we will cast the objects. Doing so we will have the db layer better abstracted, instead of having db.get around all the code

**Pull request merged**